### PR TITLE
usi-dock: add PCB tag in uuid to distinguish different PCB versions

### PR DESCRIPTION
**Pull request by victor-cheng**
From the next version of the FW, a new GUID will be used for differentiation.

### CODE REVIEW

File: plugins/usi-dock/fu-usi-dock-dmc-device.c [hunk 1]
```diff
@@ -55,6 +58,28 @@ fu_usi_dock_dmc_device_parent_notify_cb(FuDevice *device, GParamSpec *pspec, gpo
 			return;
 		}
 
+		/* allow matching PCB version */
+		serialnum = fu_device_get_serial(device);
+		if (strlen(serialnum) <= USI_SN_NUMBER_LENTH) {
```
Comment (hughsie) [inline plugins/usi-dock/fu-usi-dock-dmc-device.c:63]: you need to test that the serial number is long enough to do `serialnum[7]`, e.g. `if (strlen(serialnum) >= 8) {`

File: plugins/usi-dock/fu-usi-dock-dmc-device.c [hunk 2]
```diff
@@ -14,12 +14,15 @@ struct _FuUsiDockDmcDevice {
...
 
+#define USI_SN_NUMBER_LENTH 14
+
 static void
 fu_usi_dock_dmc_device_parent_notify_cb(FuDevice *device, GParamSpec *pspec, gpointer user_data)
 {
 	FuDevice *parent = fu_device_get_parent(device);
 	if (parent != NULL) {
 		g_autoptr(GError) error = NULL;
+		g_autofree const gchar *serialnum = NULL;
```
Comment (superm1) [inline plugins/usi-dock/fu-usi-dock-dmc-device.c:25]: you shouldn't be freeing a const value.

**Review by hughsie — APPROVED** [review]

**Comment by hughsie** [comment]
@victor-cheng

    ../plugins/usi-dock/fu-usi-dock-dmc-device.c:60:27: error: assignment discards ‘const’ qualifier from pointer target type [-Werror=discarded-qualifiers]
   60 |                 serialnum = fu_device_get_serial(device);

**Comment by superm1** [comment]
> @victor-cheng
> 
>     ../plugins/usi-dock/fu-usi-dock-dmc-device.c:60:27: error: assignment discards ‘const’ qualifier from pointer target type [-Werror=discarded-qualifiers]
>    60 |                 serialnum = fu_device_get_serial(device);
> 

It shouldn't be auto freed. It should be const, that was the same thing I said yesterday.

File: plugins/usi-dock/fu-usi-dock-dmc-device.c [hunk 3]
```diff
@@ -55,6 +56,28 @@ fu_usi_dock_dmc_device_parent_notify_cb(FuDevice *device, GParamSpec *pspec, gpo
 			return;
 		}
 
+		/* allow matching PCB version */
+		serialnum = fu_device_get_serial(device);
+		if (strlen(serialnum) >= 8) {
```
Comment (hughsie) [inline plugins/usi-dock/fu-usi-dock-dmc-device.c:61]: I think this is going to explode if the serial is ever NULL. You need `if (serialnum != NULL && strlen(serialnum) >= 8) {`

File: plugins/usi-dock/fu-usi-dock-dmc-device.c [hunk 4]
```diff
@@ -20,6 +20,7 @@ fu_usi_dock_dmc_device_parent_notify_cb(FuDevice *device, GParamSpec *pspec, gpo
 	FuDevice *parent = fu_device_get_parent(device);
 	if (parent != NULL) {
 		g_autoptr(GError) error = NULL;
+		const gchar *serialnum = NULL;
```
Comment (hughsie) [inline plugins/usi-dock/fu-usi-dock-dmc-device.c:23]: no need for `= NULL`

**Review by hughsie — APPROVED** [review]

**Pull request merged**